src/ProjectSystemTools/ProjectSystemToolsPackage.cs

```diff
@@ -152,5 +153,37 @@ private void LogRoslynWorkspaceStructure(object sender, EventArgs e)
             RoslynLogging.RoslynWorkspaceStructureLogger.ShowSaveDialogAndLog(ServiceProvider);
         }
 
+        public override IVsAsyncToolWindowFactory GetAsyncToolWindowFactory(Guid toolWindowType)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (toolWindowType == typeof(MessageListToolWindow).GUID ||
+                toolWindowType == typeof(BuildLoggingToolWindow).GUID)
+            {
+                return this;
+            }
+
+            return base.GetAsyncToolWindowFactory(toolWindowType);
+        }
+
+        protected override string GetToolWindowTitle(Type toolWindowType, int id)
+        {
+            if (toolWindowType == typeof(MessageListToolWindow))
+            {
+                return BinaryLogEditorResources.MessageListWindowTitle;
+            }
+
+            if (toolWindowType == typeof(BuildLoggingToolWindow))
+            {
+                return BuildLoggingResources.BuildLogWindowTitle;
+            }
+
+            return base.GetToolWindowTitle(toolWindowType, id);
+        }
```

Comment:
nit. I see some pattern matching potential, so I gotta comment like a dork. 😝
```suggestion
        protected override string GetToolWindowTitle(Type toolWindowType, int id) => toolWindowType switch
        {
            MessageListToolWindow _ => BinaryLogEditorResources.MessageListWindowTitle,
            BuildLoggingToolWindow _ => BuildLoggingResources.BuildLogWindowTitle,
            _ => base.GetToolWindowTitle(toolWindowType, id)
        };
```

Reply:
This is not like-for-like. `toolWindowType` is a `Type`, not a value of type `MessageListToolWindow`.

In fact I don't think this will compile, as the switch contains unreachable patterns.